src/PrestaShopBundle/Resources/views/Admin/Improve/Payment/Preferences/Blocks/payment_preferences_separated_form_block.html.twig

```diff
@@ -0,0 +1,126 @@
+{#**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ *#}
+{% form_theme paymentCurrencyRestrictionsForm '@PrestaShop/Admin/TwigTemplateForm/prestashop_ui_kit.html.twig' %}
+{% form_theme paymentGroupRestrictionsForm '@PrestaShop/Admin/TwigTemplateForm/prestashop_ui_kit.html.twig' %}
+{% form_theme paymentCountryRestrictionsForm '@PrestaShop/Admin/TwigTemplateForm/prestashop_ui_kit.html.twig' %}
+{% form_theme paymentCarrierRestrictionsForm '@PrestaShop/Admin/TwigTemplateForm/prestashop_ui_kit.html.twig' %}
+
+{% block payment_preferences_form %}
+  {% block payment_modules_currency_restrictions %}
+    {{ form_start(paymentCurrencyRestrictionsForm, {'action': path('admin_payment_currency_restrictions_save')}) }}
+    <div class="card">
+      <h3 class="card-header">
+        <i class="material-icons">euro_symbol</i>
+        {{ form_label(paymentCurrencyRestrictionsForm.currency_restrictions) }}
+      </h3>
+      <div class="card-body">
+        <div class="form-group">
+          {{ form_widget(paymentCurrencyRestrictionsForm.currency_restrictions) }}
+        </div>
+      </div>
```

Comment:
There's an issue with your integration, if a module adds an element to form it will be out of the card body which will probably mess the design.

All the restriction forms are built the same way, so you should take advantage of the form theme here:
- `PaymentModuleRestrictionsParentType` should have a fixed `getBlockPrefix` method to return `payment_module_restriction`
- define a `payment_restriction_theme.html.twig` form theme that you apply to those four form types
- in this form you can override the `payment_module_restriction_row`, this block will include the `card-header` and the `card-body` and inside this body you can simply call `form_widget(form)` which will render all the children at the right place
- you'll probably have to "move" the label, it should be on the `PaymentModuleRestrictionsParentType` form so you can use it in the `card-header`, and you may need to set the one for the restriction table to `false` or `null` if you don't want an additional label in double with the card one

Reply:
Once you move all this logic to a theme your layout here will be significantly reduced to:
```twig
{% block payment_modules_currency_restrictions %}
    {{ form_start(paymentCurrencyRestrictionsForm, {'action': path('admin_payment_currency_restrictions_save')}) }}
        {{ form_row(paymentCurrencyRestrictionsForm) }}
    {{ form_end(paymentCurrencyRestrictionsForm) }}
{% endblock %}
```